src/MSBuildTaskHost/MSBuildTaskHost.csproj

```diff
@@ -205,4 +205,8 @@
     <PackageReference Include="SourceLink.Create.CommandLine" Version="2.1.2" />
     <PackageReference Include="LargeAddressAware" Version="$(LargeAddressAwareVersion)" PrivateAssets="All" />
   </ItemGroup>
+  <ItemGroup Condition="'$(TargetFrameworkIdentifier)' != '.NETFramework'">
+    <!-- Need Win32 API on .NET Core to ping registry to determine long path support -->
+    <PackageReference Include="Microsoft.Win32.Registry" Version="4.3.0" Condition="'$(OsEnvironment)'=='windows'" />
```

Comment:
Actually, looks like this is in enough places to just reference it everywhere by putting it in [`src/Directory.Build.props`](https://github.com/Microsoft/msbuild/blob/7f95dc1575d19b9fac4da14b6f754aefe9d73a12/src/Directory.Build.props).